src/lit-html.ts

```diff
@@ -106,15 +118,33 @@ export class TemplatePart {
   }
 }
 
+
 export class Template {
   parts: TemplatePart[] = [];
   element: HTMLTemplateElement;
 
   constructor(strings: TemplateStringsArray) {
     this.element = document.createElement('template');
-    this.element.innerHTML = strings.join(exprMarker);
+    this.element.innerHTML = (() => {
+      const l = strings.length;
+      const a = new Array((l * 2) - 1);
+      let isTextBinding = false;
+      for (let i = 0; i < l; i++) {
+        const s = strings[i];
+        a.push(s);
+        if (i < l - 1) {
```

Comment:
micro-optimization: you can avoid this check by looping up to `l - 1`:
```js
for (let i = 0; i < l - 1; i++) {
  const s = strings[i];
  a.push(s);
  // We're in a text position ...
  // ...
}
a.push(strings[l -1]);

Reply:
nice :) done